com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs

```diff
@@ -472,6 +472,16 @@ public void ToSubGraph()
 
             path = EditorUtility.SaveFilePanelInProject("Save Sub Graph", "New Shader Sub Graph", ShaderSubGraphImporter.Extension, "", path);
             path = path.Replace(Application.dataPath, "Assets");
+
+            // Friendly warning that the user is generating a subgraph that would overwrite the one they are currently working on.
+            if (AssetDatabase.AssetPathToGUID(path) == selectedGuid)
+            {             
+                if (!EditorUtility.DisplayDialog("Overwrite Current Subgraph", "You are currently working on this Subgraph, are you sure you want to overwrite?", "Yes", "Cancel"))
```

Comment:
Do we want to also explicitly state that overwriting the current subgraph will destroy data?

Reply:
The 'Save As' provides an overwrite prompt already. I think the second one here is probably enough to make the point. If you have specific language you think would be best I'm happy to change it to use that. I do think 'destroy data' is a rich concept with regards to language and context, whereas 'overwrite' and 'cannot be undone' is commonly understood enough to capture that idea.

I'll go ahead and add a 'cannot be undone' disclaimer.

Reply:
Sharlene gave me some much better language to use.